osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs

```diff
@@ -100,5 +102,40 @@ public override (Vector2 position, double time) GetSnappedPosition(Vector2 posit
 
             return (snappedPosition, snappedTime);
         }
+
+        private class Ring : CircularProgress
+        {
+            [Resolved]
+            private IDistanceSnapProvider snapProvider { get; set; }
+
+            [Resolved(canBeNull: true)]
+            private EditorClock editorClock { get; set; }
+
+            private readonly HitObject referenceObject;
+
+            private readonly Color4 baseColour;
+
+            public Ring(HitObject referenceObject, Color4 baseColour)
+            {
+                this.referenceObject = referenceObject;
+
+                Colour = this.baseColour = baseColour;
+
+                Current.Value = 1;
+            }
+
+            protected override void Update()
+            {
+                base.Update();
+
+                if (editorClock == null)
+                    return;
+
+                float distanceForCurrentTime = snapProvider.DurationToDistance(referenceObject, editorClock.CurrentTime - referenceObject.GetEndTime());
+                float timeBasedAlpha = 1 - Math.Clamp(Math.Abs(distanceForCurrentTime - Size.X / 2) / 30, 0, 1);
```

Comment:
I don't think this is accounting for the distance spacing multiplier at all? Like this is what I'm seeing when I adjust the multiplier up and down while not changing time:

https://user-images.githubusercontent.com/20418176/167208952-df258fc2-1d93-4524-b284-f9fe37c7739f.mp4

I'm pretty sure the highlighted line shouldn't be shifting around like that, I'm not moving in time after all. Shouldn't this be something along the lines of

```suggestion
                float distanceForCurrentTime = (float)(snapProvider.DurationToDistance(referenceObject, editorClock.CurrentTime - referenceObject.GetEndTime()) * snapProvider.DistanceSpacingMultiplier.Value);
```

which yields the following?:

https://user-images.githubusercontent.com/20418176/167209011-309a04e3-6f0d-4ee2-a0e6-de21fdd3da96.mp4

Note that the highlighted ring does not change, it only expands and contracts (and that seems correct, I'm not moving in time after all).

Reply:
Good find, should've tested this well...

Reply:
Looks like this broke in my refactor clean up efforts.